File: Metodo de Insercion.py
tam = int(input("Cuantos numeros decea ordenar: "))
print(tam)
numero = list(range(tam))
for k in range (0,tam):
    print("Introduce un numero ",k+1)
    numero[k] = int(input())
print(numero)
for i in range(0,len(numero),1):
        pos=i
        aux=numero[i]
        while (pos>0) and (numero[pos-1]>aux):
               numero[pos]= numero[pos-1]
               pos-=1
        numero[pos]=aux
print("Escoja ordenar los numeros de manera ascendente=1, Descendente =2")
opc=int(input())
if opc==2:
        for i in range(len(numero)-1,-1,-1):
                    print(numero[i])
else:
    if opc==1:
            for i in range(0,len(numero),1):
                    print(numero[i])

File: Metodo de Insercion.cpp
#include <cstdlib>
#include <iostream>
using namespace std;
int main(int argc, char** argv) {
	int opc,pos,aux;
	int intercambio, tam;
	cout << "Cuantos numeros decea ordenar" << endl;
	cin >> tam;
	int numero[tam];
	for (int i = 0; i < tam; i++) {
		cout << "Ingrese el numeros--> " << i + 1 << endl;
		cin >> numero[i];}
	for(int i=0;i<tam;i++){
		pos=i;
		aux=numero[i];
		while((pos>0) && (numero[pos-1]>aux)){
			numero[pos] = numero[pos-1];
			pos--;}
		numero[pos]= aux;}
	cout << "Escoja ordenar los numeros de manera ascendente=1, Descendente =2" << endl;
	cin >> opc;
	switch (opc) {
	case 1: {
		for (int i = 0; i < tam; i++){
			cout << numero[i] << " ";}
		break;	}
	case 2: {
		for (int i = tam-1; i >= 0; i--){
			cout << numero[i] << " ";}
		break;}	}
	return 0;}



